include/secp256k1_extrakeys.h

```diff
@@ -0,0 +1,232 @@
+ *
+ *  Note that because the resulting point may have an odd Y coordinate, it can
+ *  not be represented by an x-only pubkey. Instead, the output_pubkey is a
+ *  normal secp256k1_pubkey.
+ *
+ *  Returns: 0 if the arguments are invalid or the resulting public key would be
+ *           invalid (only when the tweak is the negation of the corresponding
+ *           secret key). 1 otherwise.
+ *
+ *  Args:           ctx: pointer to a context object initialized for validation
+ *                       (cannot be NULL)
+ *  Out:  output_pubkey: pointer to a public key to store the result (cannot be
+ *                       NULL)
+ *  In: internal_pubkey: pointer to an x-only pubkey to apply the tweak to. Will
+ *                       be set to an invalid value if this function returns 0
+ *                       (cannot be NULL).
+ *              tweak32: pointer to a 32-byte tweak. If the tweak is invalid
+ *                       according to secp256k1_ec_seckey_verify, this function
+ *                       returns 0. For uniformly random 32-byte arrays the
+ *                       chance of being invalid is negligible (around 1 in
+ *                       2^128) (cannot be NULL).
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_xonly_pubkey_tweak_add(
+    const secp256k1_context* ctx,
+    secp256k1_pubkey *output_pubkey,
+    const secp256k1_xonly_pubkey *internal_pubkey,
+    const unsigned char *tweak32
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Tests that output_pubkey is the result of calling
+ *  secp256k1_xonly_pubkey_tweak_add with internal_pubkey and tweak32.
+ *
+ *  The output_pubkey is represented by its 32-byte x-only serialization and its
+ *  pk_parity, which can both be obtained by converting the result of tweak_add
+ *  to a secp256k1_xonly_pubkey.
+ *
+ *  Note that this alone does _not_ verify that output_pubkey is a commitment.
+ *  If the tweak is not chosen in a specific way, the output_pubkey can easily
+ *  be the result of a different internal_pubkey and tweak.
+ *
+ *  Returns: 0 if the arguments are invalid or the output_pubkey is not the
+ *           result of tweaking the internal_pubkey with tweak32. 1 otherwise.
+ *  Args:           ctx: pointer to a context object initialized for validation
+ *                       (cannot be NULL)
+ *  In: output_pubkey32: pointer to a serialized xonly_pubkey (cannot be NULL)
+ *     output_pk_parity: the pk_parity value that is returned when
+ *                       converting the output_pubkey of
+ *                       secp256k1_xonly_pubkey_tweak_add to an xonly_pubkey
+ *      internal_pubkey: pointer to an x-only public key object to apply the
+ *                       tweak to (cannot be NULL)
+ *              tweak32: pointer to a 32-byte tweak (cannot be NULL)
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_xonly_pubkey_tweak_add_test(
```

Comment:
Maybe `test` is not a great name because we only use this to refer functional tests so far, and it's somewhat confusing. Suggestions: `check`, `verify`. Maybe `secp256k1_xonly_pubkey_tweak_add_check` or even `secp256k1_xonly_pubkey_check_tweak_add`, which I like slightly more. That's still our `datatype_operation` style because tweak is not really a datatype. This is used in a function `CheckPayToContract` in https://github.com/bitcoin/bitcoin/pull/17977/commits/a9f418aeb0d31acadd0d5114158fb1fcb9094f40#diff-2c33d0ff0ed53b0902e4af9a745a152fR181 by the way.


Sorry for nitpicking this single thing right now without anything else but I stumbled upon this just now and probably will forget about otherwise.

Reply:
It was originally `verify` and I didn't like it because it could suggest that it actually verifies the commitment, which it isn't https://github.com/bitcoin-core/secp256k1/pull/558#discussion_r347126158

Reply:
Oh I see. I think `check` is then really the best.

Reply:
Renamed to `secp256k1_xonly_pubkey_tweak_add_check`